arches_querysets/datatypes/resource_types.py

```diff
@@ -0,0 +1,181 @@
+        try:
+            rxrs = tile.resourceinstance.filtered_from_resxres
+        except:
+            if arches_version >= (8, 0):  # TODO: why?
+                rxrs = tile.resourceinstance.from_resxres.all()
+            else:
+                rxrs = tile.resourceinstance.resxres_resource_instance_ids_from.all()
+
+        def handle_missing_data(to_resource_id):
+            msg = f"Missing ResourceXResource target: {to_resource_id}"
+            logger.warning(msg)
+
+        for inner_val in value:
+            if not inner_val:
+                continue
+            for rxr in rxrs:
+                to_resource_id = (
+                    rxr.resourceinstanceidto_id
+                    if arches_version < (8, 0)
+                    else rxr.to_resource_id
+                )
+                if to_resource_id == uuid.UUID(inner_val["resourceId"]):
+                    try:
+                        to_resource = (
+                            rxr.resourceinstanceidto
+                            if arches_version < (8, 0)
+                            else rxr.to_resource
+                        )
+                    except models.ResourceInstance.DoesNotExist:
+                        handle_missing_data(to_resource_id)
+                        break
+                    if to_resource is None:
+                        handle_missing_data(to_resource_id)
+                        break
+                    related_resources.append(to_resource)
+                    break
+
+        return related_resources
+
+    def get_details(self, tile, node):
+        lang = get_language()
+        value = tile.data.get(str(node.nodeid)) or []
+        related_resources_by_id = {
+            rr.pk: rr for rr in self.get_related_resources(tile, value)
+        }
+        ret = []
+        for inner_val in value:
+            if not inner_val:
+                continue
+            if related := related_resources_by_id.get(
+                uuid.UUID(inner_val["resourceId"]), None
+            ):
+                ret.append(
+                    {
+                        "resource_id": str(related.pk),
+                        # TODO: gracefully handle missing language.
+                        "display_value": related.descriptors[lang]["name"],
+                    }
+                )
+            else:
+                ret.append(
+                    {
+                        "resource_id": str(related.pk),
+                        "display_value": _("Missing"),
+                    }
+                )
+        return ret
+
+    def get_interchange_value(self, value, *, details=None, **kwargs):
+        if not value:
+            return None
+        if details is None:
+            details = self.get_details(value)
+        return details[0]["resource_id"]
+
+
+class ResourceInstanceListDataType(ResourceInstanceDataType):
+    def collects_multiple_values(self):
+        return True
+
+    def to_python(self, value, *, tile, **kwargs):
```

Comment:
Am I understanding this right? If a resource list has no related resource it doesn't get returned?

Reply:
I think we're missing each other maybe? 

The call to `ResourceInstanceDataType.get_related_resources(tile, value)` is what I'm getting hung up on. It's possible to have a `ResourceInstanceListDataType` that has a `ResourceInstanceDataType` that has no resource relations, right?

In that case this would return none